letsencrypt-nginx/letsencrypt_nginx/configurator.py

```diff
@@ -362,6 +362,26 @@ def enhance(self, domain, enhancement, options=None):
         except errors.PluginError:
             logger.warn("Failed %s for %s", enhancement, domain)
 
+    def _enable_redirect(self, vhost, unused_options):
+        """Redirect all equivalent HTTP traffic to ssl_vhost.
+
+        Add rewrite directove to non https traffic
+
+        .. note:: This function saves the configuration
+
+        :param vhost: Destination of traffic, an ssl enabled vhost
+        :type vhost: :class:`~letsencrypt_apache.obj.VirtualHost`
+
+        :param unused_options: Not currently used
+        :type unused_options: Not Available
+        """
+        redirect_block = [[['if', '($scheme != "https")'],
```

Comment:
This works fine, but a more efficient way would be `return 301 https://$host$uri;`. That avoids firing up the comparatively expensive regex machinery (and, for my reference, has been supported since 0.7.51, released in 2009).

We might be able to do even better in the future if we make a fresh vhost for the HTTP side:
```
server {
    server_name example.com;
    return 301 https://$host$request_uri;
}
```
That eliminates even the `if` but would require a bit of redesign.